api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -337,9 +337,13 @@ private void discontinueExistingOrdersIfNecessary(Order order, Boolean isRetrosp
 			return;
 		}
 
+		Date asOfDate = null;
+		if (isRetrospective)
+			asOfDate = order.getDateActivated();
+
 		//Mark first order found corresponding to this DC order as discontinued.
 		List<? extends Order> orders = getActiveOrders(order.getPatient(), order.getOrderType(), order.getCareSetting(),
-		    null);
+		    asOfDate);
```

Comment:
I think i would disagree with doing this, discontinuing existing  orders is any only makes when doing real time order entry